packages/radix-vue/src/Tree/TreeItem.vue

```diff
@@ -0,0 +1,173 @@
+<script lang="ts">
+export interface TreeItemProps<T> extends PrimitiveProps {
+  /** Value given to this item */
+  value: T
+  /** Level of depth */
+  level: number
+}
+
+export type SelectEvent<T> = CustomEvent<{ originalEvent: PointerEvent, value?: T, isExpanded: boolean, isSelected: boolean }>
+export type ToggleEvent<T> = CustomEvent<{ originalEvent: PointerEvent, value?: T, isExpanded: boolean, isSelected: boolean }>
+
+export type TreeItemEmits<T> = {
+  /** Event handler called when the selecting item. <br> It can be prevented by calling `event.preventDefault`. */
+  select: [event: SelectEvent<T>]
+  /** Event handler called when the selecting item. <br> It can be prevented by calling `event.preventDefault`. */
+  toggle: [event: ToggleEvent<T>]
+}
+
+const TREE_SELECT = 'tree.select'
+</script>
+
+<script setup lang="ts" generic="T extends Record<string, any>">
+import { Primitive, type PrimitiveProps } from '@/Primitive'
+import { RovingFocusItem } from '@/RovingFocus'
+import { injectTreeRootContext } from './TreeRoot.vue'
+import { computed } from 'vue'
+import { useCollection } from '@/Collection'
+import { handleAndDispatchCustomEvent, useForwardRef } from '@/shared'
+import { flatten } from './utils'
+
+defineOptions({
+  inheritAttrs: false,
+})
+
+const props = withDefaults(defineProps<TreeItemProps<T>>(), {
+  as: 'li',
```

Comment:
Maybe not default to `li`? This makes the default markup [invalid](https://github.com/radix-vue/radix-vue/blob/48f38b09828a7a4c3654767b2e6f285a468a8508/packages/radix-vue/src/Tree/__snapshots__/Tree.test.ts.snap#L4) (`TreeRoot` is a `div`, `TreeItem` is a `li`). Setting `TreeRoot` to `ul`/`ol` would not work with `TreeVirtualizer` (it renders a spurios `div` resulting in `ul > div > li`).

Since this implementation flattens the tree in HTML, using a `ul`/`ol` is not that helpful anyway. One would expect a nested list of lists structure to semantically represent the tree.

Reply:
Thanks for the comment @d--j ! 💚

I've changed `TreeRoot` to be `ul`, and if user is using virtualization they can change the rendered element to `div`/`ul`/`ol` based on their needs.

And the examples we have are using flattened tree (as this will supports virtualization and dnd easily), I'll add example to showcase proper nested list  😁

Reply:
Added a story here https://github.com/radix-vue/radix-vue/pull/976/files#diff-375bb0c69ffc5c2745835f594c74785416bc92517dc35f2e9061688e5490c659

Will include a stackblitz in the docs just in case anyone is looking for it 😁